library/common/engine.cc

```diff
@@ -59,10 +68,13 @@ envoy_status_t Engine::main(const std::string config, const std::string log_leve
 
       cv_.notifyAll();
     } catch (const Envoy::NoServingException& e) {
+      free(composed_config);
```

Comment:
the above comment would also save us from these frees :)

Reply:
Honestly we don't need them at all with the PANIC happening below, but I put it here in case we change the behavior again. Actually now that I think about it, I handle the memory via a unique_ptr so I'll do that.

Reply:
Updated.